packages/cron/test/cargo.spec.js

```diff
@@ -181,15 +181,28 @@ describe('Fix dag sizes migration', () => {
     assert.strictEqual(toBeUpdated.length, updatedCids.length)
   })
 
-  it('updates the updated_at field', async () => {
-    const { allUploadsBefore, allUploadsAfter } = await updateDagSizesWrp({ user })
+  it('updates only when needed', async () => {
+    const toBeUpdated = contentItems
+      .filter((c) => getToBeUpdatedNull(c) || getToBeUpdatedWrong(c))
+      .map(c => c.cid)
+
+    const beforeContents = await getContents(dbClient, toBeUpdated)
+    await updateDagSizesWrp({ user })
+    const afterFirstRoundContents = await getContents(dbClient, toBeUpdated)
+
+    const updatedCidsFirstRound = afterFirstRoundContents.filter((cAfter) => {
+      const beforeContent = beforeContents.find((cBefore) => cAfter.cid === cBefore.cid)
+      return beforeContent?.updated_at !== cAfter.updated_at
+    })
+    assert.strictEqual(updatedCidsFirstRound.length, toBeUpdated.length)
```

Comment:
Would it be better to compare the actual CIDs here, to ensure that the correct items were updated, rather than just checking that the correct number were updated?
```suggestion
    toBeUpdatedCids = toBeUpdated.map(c => ci.cid)
    assert.strictEqual(updatedCidsFirstRound.slice().sort(), toBeUpdatedCids.slice().sort())
```

Reply:
Nice one!